features/dd-sdk-android-trace/src/test/kotlin/com/datadog/trace/core/CoreTracerTest.kt

```diff
@@ -0,0 +1,235 @@
+        // Given
+        val properties = Properties().apply {
+            setProperty(TracerConfig.SERVICE_MAPPING, baggageItem)
+            setProperty(TracerConfig.SPAN_TAGS, baggageItem)
+            setProperty(TracerConfig.HEADER_TAGS, baggageItem)
+        }
+
+        // When
+        val config = createInstanceWithPrivateConstructor(
+        Config::class.java,
+            ConfigProvider.withPropertiesOverride(properties)
+        )
+        val tracer = tracerBuilder().config(config).build()
+
+        // Then
+        assertThat(expectedMapping).isEqualTo(tracer.captureTraceConfig().serviceMapping)
+    }
+
+    @ParameterizedTest
+    @MethodSource("baggageMapping")
+    fun `verify mapping configs on tracer`(baggageItem: String, expectedMapping: Map<String, String>) {
+        // Given
+        val properties = Properties().apply {
+            setProperty(TracerConfig.BAGGAGE_MAPPING, baggageItem)
+        }
+        val config = createInstanceWithPrivateConstructor(
+        Config::class.java,
+            ConfigProvider.withPropertiesOverride(properties)
+        )
+
+        // When
+        val tracer = tracerBuilder().config(config).build()
+
+        // Then
+        assertThat(expectedMapping).isEqualTo(tracer.captureTraceConfig().baggageMapping)
+    }
+
+    @Test
+    fun `root tags are applied only to root spans`() {
+        // Given
+        val tracer = tracerBuilder().localRootSpanTags(mapOf("only_root" to "value")).build()
+        val root = tracer.buildSpan(instrumentationName, "my_root").start()
+
+        // When
+        val child = tracer.buildSpan(instrumentationName, "my_child").asChildOf(root.context()).start()
+
+        // Then
+        assertThat((root.context() as DDSpanContext).tags).containsKey("only_root")
+        assertThat((child.context() as DDSpanContext).tags).doesNotContainKey("only_root")
+
+        // Tear down
+        child.finish()
+        root.finish()
+        tracer.close()
+    }
+
+    @Test
+    fun `priority sampling when span finishes`() {
+        // Given
+        val writer = ListWriter()
+        val tracer = tracerBuilder().writer(writer).build()
+
+        // When
+        val span = tracer.buildSpan(instrumentationName, "operation").start()
+        span.finish()
+        writer.waitForTraces(1)
+
+        // Then
+        assertThat(span.samplingPriority).isEqualTo(PrioritySampling.SAMPLER_KEEP.toInt())
+
+        // Tear down
+        tracer.close()
+    }
+
+    @Test
+    fun `priority sampling set when child span complete`() {
+        // Given
+        val writer = ListWriter()
+        val tracer = tracerBuilder().writer(writer).build()
+
+        // When
+        val root = tracer.buildSpan(instrumentationName, "operation").start()
+        val child = tracer.buildSpan(instrumentationName, "my_child").asChildOf(root.context()).start()
+        root.finish()
+
+        // Then
+        assert(child.samplingPriority == null)
```

Comment:
Java code returns NULL instead of Int? and that code crashes with an NPE, I could cast it but I prefer it this way. Same for the other comment on the same topic.

Reply:
Is it possible to use `check` then if using `assertThat` brings the problem here (not sure though still where exactly the problem lies)?

The thing this line will alway pass unless JVM is started with `-ea` flag (enable assertions), because this `assert(...)` is shortcut for the Java's `assert ....` (https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html).

Event Kotlin's `assert` docs says:

> Throws an [AssertionError] if the [value] is false and runtime assertions have been enabled on the JVM using the *-ea* JVM option.

Reply:
actually was an issue in the Java code, that method was not marked as @Nullable and Kotlin was not returning an Int?